src/include/function/cast/cast_functions.h

```diff
@@ -87,27 +88,81 @@ inline std::string CastToString::castToStringWithVector(
 
 template<typename SRC, typename DST>
 static inline void numericDownCast(SRC& input, DST& result, const std::string& dstTypeStr) {
-    if (input < std::numeric_limits<DST>::min() || input > std::numeric_limits<DST>::max()) {
+    if ((signed)input < std::numeric_limits<DST>::min() ||
+        (signed)input > std::numeric_limits<DST>::max()) {
+        auto a = std::numeric_limits<DST>::min();
+        auto b = std::numeric_limits<DST>::max();
         throw common::RuntimeException(
             "Cast failed. " + std::to_string(input) + " is not in " + dstTypeStr + " range.");
     }
     result = (DST)input;
 }
 
+template<typename T>
+static inline bool tryCastStringToNum(const char* input, const uint64_t& len, T& result) {
+    // skip leading spaces
+    uint64_t start = 0;
```

Comment:
Can you refactor the logic of removing leading/trailing spaces to a function inside stringUtils.h? Since it is also used in other places (e.g. parsing struct, list).
```
char* removeSpaces(char* input, const uint64_t& len);
```
The function should return the a pointer to the new string.